samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -43,53 +48,85 @@
 import static org.mockito.Mockito.when;
 
 public class TestStreamProcessor {
+  private Map<String, Boolean> processorListenerState;
+  private static final String ON_START = "onStart";
+  private static final String ON_SHUTDOWN = "onShutdown";
+  private static final String ON_FAILURE = "onFailure";
+
```

Comment:
An enum would be more suitable here because it nicely groups the possible states.